practices/tdd/readme.md

```diff
@@ -44,6 +44,44 @@ So what are some of the benefits that TDD provides to make this a practice worth
+
+#### ***What should the names of the tests be?***
+
+**Name test classes similarly to the classes they're testing.** An easy way to do this is to use the same name as the implementation class with "Test" at the end. So if you have a "Calculator" class, the test class would be called "CalculatorTest".
+
+**Use descriptive names for test methods.** It's helpful to name test methods based on the condition being tested and the expected result. This makes it easier to understand exactly what the test is doing. For example, if you have a method named "Add" that is expected to increment a variable named "Value" when it is called, the test method for "Add" might be named "WhenAddThenValueIsIncremented".
+
+#### ***How should the tests be written?***
+
+**Arrange, Act, Assert.** Use this pattern as the structure for your test. In the "Arrange" step you set up the testing objects and prepare the prerequisites for your test. The "Act" step is where the actual work of the test is performed. And finally, the "Assert" step is where the result of the test is verified. This structure can be seen in all of the examples we've written below.
+
+**Write assertions first.** This makes the purpose of the test clear from the beginning.
+
+**Minimize assertions in each test method.** Testing several assertions at once can make it difficult to determine which one is causing the test to fail in the future. This doesn't mean that tests need to be limited to one assertion, though. If there are other assertions that test the same logical condition it can make sense to use them in the same test.
+
+**Use mocks when appropriate.** In testing, mocking replicates the behavior of a service, object, or process. This can be useful when a method that is being tested has dependencies on external resources such as databases. Mocking those external dependencies makes it so that you're just testing the code in the method and not in the external dependencies. This means that if the test fails, you can know that it's happening inside your method and not elsewhere. Mocking external resources can also speed up tests because it eliminates the latency in communication with those resources.
+
+Using too many mocks can be a problem too, however. Using a lot of mocks can result in a test being tightly coupled with the implementation of the code that it's testing. This goes against a key part of TDD - that you want to be testing behaviors of your code and not it's implementation. Coupling to the implementation makes tests more susceptible to failure when refactoring occurs.
+
+There are several frameworks such as Mockito and Moq that aid with mocking. You can see how these are implemented in the examples below.
```

Comment:
@rebecca-gardner these are specific to java and C# yes? might just be worth saying that there are many mocking libraries for a variety of languages and frameworks, or something to that affect. To me, this slightly implies those are inherently good, since they are the only ones mentioned